libvirt.go

```diff
@@ -147,21 +148,35 @@ func (d *Driver) GetURL() (string, error) {
 }
 
 func (d *Driver) getConn() (*libvirt.Connect, error) {
-	if d.conn == nil {
-		conn, err := libvirt.NewConnect(connectionString)
+	log.Debugf("Connecting to session")
+	if d.sessionConn == nil {
+		conn, err := libvirt.NewConnect(sessionConnectionString)
+		if err != nil {
+			log.Errorf("Failed to connect to libvirt: %s", err)
+			return &libvirt.Connect{}, errors.New("Unable to connect to kvm driver")
+		}
+		d.sessionConn = conn
+	}
+	return d.sessionConn, nil
+}
+
+func (d *Driver) getSystemConn() (*libvirt.Connect, error) {
```

Comment:
If we still need getSystemConn() after this change, this means user still needs to be given access to qemu:///system by crc preflight checks. Removing this need was for me one of the big reasons for moving to qemu:///session :(

Reply:
No, this is ready-only access. See the change in this function.

Reply:
Well, read-only access still means we need access to qemu:///system. Yes this is better, and we can adjust the changes crc setup makes so that the user only has read-only access to qemu://system, but this hasn't been done yet. If it had been possible to remove all usage of qemu:///system, the change would be trivial, just remove the prelfight checks ;)

Reply:
All users get read-only access to system connection AFAIK so I'm still not quite following.

Reply:
if you're talking about `crc`, that still needs write-access to system connection cause otherwise how would it create the network there.

Reply:
but only for pre-flight fixing. After that, it doesn't.